arch/arm64/include/asm/kfence.h

```diff
@@ -17,17 +13,7 @@
  * default, however, we do not have struct pages for static allocations.
  */
 
-static inline bool arch_kfence_initialize_pool(void)
-{
-	const unsigned int num_pages = ilog2(roundup_pow_of_two(KFENCE_POOL_SIZE / PAGE_SIZE));
-	struct page *pages = alloc_pages(GFP_KERNEL, num_pages);
-
-	if (!pages)
-		return false;
-
-	__kfence_pool = page_address(pages);
-	return true;
-}
+static inline bool arch_kfence_initialize_pool(void) { return true; }
```

Comment:
We probably need to declare the same stub for x86?

Reply:
The function name didn't change. For x86, arch_kfence_initialize_pool() still has to do the set_memory_4k stuff on the static pool. 

Or maybe I missunderstood?

The way it should work:
- If we use the static pool, the generic alloc_kfence_pool() in core.c does nothing, as is the case for x86 right now.

- If we use the dynamic pool, one of 3 things can happen:
  1. arch_kfence_initialize_pool() is empty, and alloc_kfence_pool() just does the work.
  2. arch_kfence_initialize_pool() uses alloc_kfence_pool() (since it's included, it can declare the static function and just use it) and does some extra stuff.
  3. arch_kfence_initialize_pool() does something entirely different, and alloc_kfence_pool() will do nothing since __kfence_pool is non-NULL.

If we switch x86 over to a dynamic pool and give up on the static pool,I think it'll be option (ii) for x86.

Reply:
Ah, I've been staring at that function, but failed to understand it has the same name. Sorry about that!